File: app/src/main/java/com/aquamorph/frcmanager/activities/TeamSummary.java
package com.aquamorph.frcmanager.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.aquamorph.frcmanager.R;
import com.aquamorph.frcmanager.fragments.TeamScheduleFragment;

/**
 * Activiy with a summary of the schedule for a given team number.
 *
 * @author Christian Colglazier
 * @version 3/29/2016
 */
public class TeamSummary extends AppCompatActivity {

	private String teamNumber = "";

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.settings);

		Bundle extras = getIntent().getExtras();
		if (extras != null) {
			teamNumber = extras.getString("teamNumber").replaceAll("[^\\d.]", "");
		}

		Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
		if (toolbar != null) {
			toolbar.setTitle(teamNumber);
		}
		setSupportActionBar(toolbar);
		getSupportActionBar().setDisplayHomeAsUpEnabled(true);
		TeamScheduleFragment teamScheduleFragment = TeamScheduleFragment.newInstance();
		teamScheduleFragment.setTeamNumber(teamNumber);
		getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, teamScheduleFragment).commit();
		MainActivity.theme(this);
	}

	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
			case android.R.id.home:
			default:
				this.finish();
				break;
		}
		return super.onOptionsItemSelected(item);
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/activities/TeamSummary.kt
package com.aquamorph.frcmanager.activities

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.aquamorph.frcmanager.R
import com.aquamorph.frcmanager.fragments.TeamScheduleFragment
import com.aquamorph.frcmanager.utils.Constants.getTeamName
import com.aquamorph.frcmanager.utils.Constants.getTeamRank
import com.aquamorph.frcmanager.utils.Constants.getTeamRecord

/**
 * Activiy with a summary of the schedule for a given team number.
 *
 * @author Christian Colglazier
 * @version 3/31/2018
 */
class TeamSummary : AppCompatActivity() {

    private var teamNumber = ""

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings)

        val extras = intent.extras
        if (extras != null) {
            teamNumber = extras.getString("teamNumber")!!.replace("[^\\d.]".toRegex(), "")
        }

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        if (toolbar != null) {
            toolbar.title = "$teamNumber - " + getTeamName("frc$teamNumber")

            if (getTeamRecord("frc$teamNumber").isNotEmpty() &&
                    getTeamRank("frc$teamNumber").isNotEmpty()) {
                toolbar.subtitle = getTeamRank("frc$teamNumber") + " " +
                        getTeamRecord("frc$teamNumber")
            }
        }
        setSupportActionBar(toolbar)
        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        }
        val teamScheduleFragment = TeamScheduleFragment.newInstance()
        teamScheduleFragment.setTeamNumber(teamNumber)
        supportFragmentManager.beginTransaction().replace(R.id.content_frame, teamScheduleFragment).commit()
        MainActivity.theme(this)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> this.finish()
        }
        return super.onOptionsItemSelected(item)
    }
}
